packages/edit-post/src/components/layout/index.js

```diff
@@ -141,15 +140,15 @@ export default compose(
 	withSelect( ( select ) => {
 		const editorSettings = select( 'core/editor' ).getEditorSettings();
 		return {
-			mode: select( 'core/edit-post' ).getEditorMode(),
+			contentEditingMode: select( 'core/edit-post' ).getEditorMode(),
 			editorSidebarOpened: select( 'core/edit-post' ).isEditorSidebarOpened(),
 			pluginSidebarOpened: select( 'core/edit-post' ).isPluginSidebarOpened(),
 			publishSidebarOpened: select( 'core/edit-post' ).isPublishSidebarOpened(),
 			hasFixedToolbar: select( 'core/edit-post' ).isFeatureActive( 'fixedToolbar' ),
 			hasActiveMetaboxes: select( 'core/edit-post' ).hasMetaBoxes(),
 			isSaving: select( 'core/edit-post' ).isSavingMetaBoxes(),
 			isRichEditingEnabled: editorSettings.richEditingEnabled,
-			isEditingTemplatePost: editorSettings.editTemplatePost,
+			viewEditingMode: editorSettings.viewEditingMode,
```

Comment:
As mentioned at https://github.com/WordPress/gutenberg/pull/16565#discussion_r305286592, can we make this to be part of state rather than as a setting of the editor? Does that make sense to you?

Reply:
Yes, the benefit would be that we could persist it between page loads which I personally find it annoying with the current approach.